src/CompletedTaskList.js

```diff
@@ -2,12 +2,46 @@ import React from "react"
 
 import CreatedTask from "./CreatedTask"
 import Header from "./Header"
+import todosData from "./todosData"
 
-export default function CompletedTaskList(props) {
-  return (
-    <div>
-      <Header title="Completed Tasks" />
-      <CreatedTask task={props.task} completed="completed" />
-    </div>
-  )
-}
\ No newline at end of file
+class CompletedTaskList extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      todos: todosData,
+    }
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  handleClick(id) {
+    this.setState((prevState) => {
+      const updatedTodos = prevState.todos.map((todo) => {
```

Comment:
You nailed the idea behind state management here! I was thinking that maybe if there was a component whose sole responsibility was to render the todo item, it could handle updating itself and then this state update wouldn't have to be that complex, maybe something like:

```jsx
class TodoItem extends React.Component {
  constructor(props) {
    super()
    this.state = {
      todoItem: props.todoItem
    }
    this.handleClick = this.handleClick.bind(this)
}

handleClick() {
  this.setState((prevState) => { 
    return { todoItem: { ...prevState.todoItem, completed: !prevState.todoItem.completed } } 
});
}

render() {
   const { todoItem } = this.state;
   return <CreatedTask 
                 key={todoItem.id} 
                 id={todoItem.id} 
                 task={todoItem.text} 
                 completed={todoItem.completed} 
                 handleClick={this.handleClick} />
};
```

Reply:
^ The way @aledustet sets this up would be the way I would recommend. Right now it looks like you are remapping all of the todos each time you update one of them. If each task was responsible for its own state, then you wouldn't have to worry about this.

Reply:
This is fabulous!  Thank you!  I'll see if I can incorporate this into the tutorial flow, but I absolutely agree that this is more manageable!